core/server/master/src/main/java/alluxio/master/file/meta/SimpleInodeLockList.java

```diff
@@ -56,43 +79,43 @@
    */
   public SimpleInodeLockList(InodeLockManager inodeLockManager) {
     mInodeLockManager = inodeLockManager;
-    mEntries = new ArrayList<>(INITIAL_CAPACITY);
-    mFirstWriteLockIndex = 0;
+    mInodes = new ArrayList<>(INITIAL_INODES_CAPACITY);
+    mLocks = new ArrayList<>(INITIAL_LOCKS_CAPACITY);
+    mFirstWriteLockIndex = NO_WRITE_LOCK_INDEX;
   }
 
   @Override
   public void lockInode(Inode inode, LockMode mode) {
     mode = nextLockMode(mode);
-    if (!mEntries.isEmpty()) {
+    if (!mLocks.isEmpty()) {
       Preconditions.checkState(!endsInInode(),
           "Cannot lock inode %s for lock list %s because the lock list already ends in an inode",
           inode.getId(), this);
-      String lastEdgeName = ((EdgeEntry) lastEntry()).getEdge().getName();
-      Preconditions.checkState(inode.getName().equals(lastEdgeName),
-          "Expected to lock inode %s but locked inode %s", lastEdgeName, inode.getName());
+      Preconditions.checkState(inode.getName().equals(mLastEdge.getName()),
```

Comment:
`mLastEdge` could be `null`, right?

Reply:
no, the previous precondition has checked that it ends in an edge, so mLastEdge must not be null.